### DISCUSSION TITLE
Inline Form Interface

### DISCUSSION THREAD

**Discussion by hanneskuettner** [directus/directus#18041] [Feature Requests]
### Summary

Create a M2O interface that renders the related item in the parent form.  First brought up in #3474

This allows easy sharing of common fields between different collections without the extra mental overhead of selecting or creating a new item in the M2O Dropdown and editing the item in a separate drawer.

### Motivation

Often we have common fields that could be shared by multiple collections, eg. a "Meta" collection that includes title, description, OpenGraph definitions, etc.. shared in the `pages`, `news` and `products` collections.

The optimal way to structure the database would be to create one common collection that is related through a M2O with other collections (though in this case it is basically a O2O). 

Currently the editing experience with the Dropdown is more cumbersome than not.

### Detailed Design

The "Inline Form" should be implemented as a separate M2O interface that allows the selection of a related collection.

The interface should be rendered "inline" as the name suggests, so seamlessly integrated into the parent item editor page.
It should display all the fields of the related item, except for a possible circular field that points to the parent item.
This can be achieved by utilizing the existing v-form component similar to how it is used in the drawer-item. 

Optionally it would be nice to group it into a details group through an interface option to better organize the editing experience of the containing form.

### Must Have:
- The ability to select a related collection
- Inline editing for a related item in a containing parent item
- Documentation

### Should Have:
- [The option to select if the related item should always be created or only if some content is created](https://github.com/directus/directus/discussions/3474#discussioncomment-327660)
- Validation errors displayed in the containing form with proper field naming, eg. "Meta / Title must be at least 100 characters"

### Could Have:
- The ability to optionally wrap the form in a details group with an icon, title and maybe a color (similar to a group)
- The ability to hide the field title if the form should be integrated seamlessly into the parent form

### Won't Have:
- 

### Drawbacks

- The interface could be implemented in a custom extension, but judging by @benhaynes reaction to the Q&A discussion it seems like it's worth including it in the core functionality

### Alternatives

The alternative is using the existing M2O dropdown, but see the drawbacks discussed above.

### Adoption Strategy

It is not a braking change but only the addition of an interface.

### Unresolved Questions

If the nested validation needs to be adjusted. Because as of some testing I did the validation for related fields works as expected but the `collection` field in the ValidationError stays empty, preventing the display of a full "path" to the field and also the pretty print (and probably translation) of the field name.

**Comment by hanneskuettner** [comment]
I've published my proposed idea as a separate extension that can be used until there is time to flush this out and bring it to the core.
You can find it here https://github.com/hanneskuettner/directus-extension-inline-form-interface

**Comment by davidpp** [comment]
Added this extension, and would already consider a must have.  This allowed me to split tons of general "settings" in a re-usable collection without end users having to be aware of the change.

**Comment by carwei** [comment]
Excellent work on both the extension and proposal 👍

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
